sgkit_vcf/vcf_reader.py

```diff
@@ -0,0 +1,286 @@
+                elif len(alleles) < n_allele:
+                    alleles = alleles + ([""] * (n_allele - len(alleles)))
+                variant_alleles.append(alleles)
+                max_variant_allele_length = max(
+                    max_variant_allele_length, max(len(x) for x in alleles)
+                )
+
+                gt = variant.genotype.array()
+                call_genotype[i] = gt[..., 0:-1]
+                call_genotype_phased[i] = gt[..., -1]
+
+            # Truncate np arrays (if last chunk is smaller than chunk_length)
+            if i + 1 < chunk_length:
+                variant_contig = variant_contig[: i + 1]
+                variant_position = variant_position[: i + 1]
+                call_genotype = call_genotype[: i + 1]
+                call_genotype_phased = call_genotype_phased[: i + 1]
+
+            variant_id = np.array(variant_ids, dtype="O")
+            variant_id_mask = variant_id == "."
+            variant_alleles = np.array(variant_alleles, dtype="O")
+
+            ds: xr.Dataset = create_genotype_call_dataset(
+                variant_contig_names=variant_contig_names,
+                variant_contig=variant_contig,
+                variant_position=variant_position,
+                variant_alleles=variant_alleles,
+                sample_id=sample_id,
+                call_genotype=call_genotype,
+                call_genotype_phased=call_genotype_phased,
+                variant_id=variant_id,
+            )
+            ds["variant_id_mask"] = (
+                [DIM_VARIANT],
+                variant_id_mask,
+            )
+            ds.attrs["max_variant_id_length"] = max_variant_id_length
+            ds.attrs["max_variant_allele_length"] = max_variant_allele_length
+
+            if first_variants_chunk:
+                # Enforce uniform chunks in the variants dimension
+                # Also chunk in the samples direction
+                encoding = dict(
+                    call_genotype=dict(chunks=(chunk_length, chunk_width, n_ploidy)),
+                    call_genotype_mask=dict(
+                        chunks=(chunk_length, chunk_width, n_ploidy)
+                    ),
+                    call_genotype_phased=dict(chunks=(chunk_length, chunk_width)),
+                    variant_allele=dict(chunks=(chunk_length, n_allele)),
+                    variant_contig=dict(chunks=(chunk_length,)),
+                    variant_id=dict(chunks=(chunk_length,)),
+                    variant_id_mask=dict(chunks=(chunk_length,)),
+                    variant_position=dict(chunks=(chunk_length,)),
+                    sample_id=dict(chunks=(chunk_width,)),
+                )
+
+                ds.to_zarr(output, mode="w", encoding=encoding)
+                first_variants_chunk = False
+            else:
+                # Append along the variants dimension
+                ds.to_zarr(output, append_dim=DIM_VARIANT)
+
+
+def vcf_to_dataset(
+    input: PathType,
+    output: PathType,
```

Comment:
It would be useful if this was a `store` argument instead with the same type as `to_zarr` so it can be passed in as a remote store (to at least avoid one local copy).  I was planning on doing that with bgen.